cares-center-inventory-Song/crud.py

```diff
@@ -0,0 +1,242 @@
+
+def delete_replenish_items_by_item_id(db: Session, item_id: str):
+    replenish_item_to_delete = db.query(models.ReplenishItem).filter(models.ReplenishItem.item_id == item_id).all()
+    db.delete(replenish_item_to_delete)
+    db.commit()
+
+def delete_replenish_items_by_replenish_id(db: Session, replenish_id: str):
+    replenish_item_to_delete = db.query(models.ReplenishItem).filter(models.ReplenishItem.replenish_id == replenish_id).all()
+    db.delete(replenish_item_to_delete)
+    db.commit()
+ 
+# transaction_history
+def create_transaction_history(db: Session, transaction_history: schemas.TransactionHistoryCreate):
+    db_transaction_history = models.TransactionHistory(
+        transaction_date = transaction_history.transaction_date,
+        transaction_points = transaction_history.transaction_points,
+        customer_id = transaction_history.customer_id,
+        manager_id = transaction_history.manager_id,
+    )
+    db.add(db_transaction_history)
+    db.commit()
+    db.refresh(db_transaction_history)
+    return db_transaction_history
+
+def get_transaction_histories_by_transaction_date(db: Session, transaction_date: str):
+    return db.query(models.TransactionHistory).filter(models.TransactionHistory.transaction_date == transaction_date).all()
+
+def get_transaction_histories_by_customer_id(db: Session, customer_id: str):
+    return db.query(models.TransactionHistory).filter(models.TransactionHistory.customer_id == customer_id).all()
+
+def  get_transaction_histories_by_transaction_date_and_customer_id(db: Session, transaction_date: str, customer_id: str):
+    return db.query(models.TransactionHistory).filter(and_(models.TransactionHistory.transaction_date == transaction_date, models.TransactionHistory.customer_id == customer_id)).all()
+
+def get_transaction_histories_by_manager_id(db: Session, manager_id: str):
+    return db.query(models.TransactionHistory).filter(models.TransactionHistory.manager_id == manager_id).all()
+
+def delete_transaction_history_by_transaction_id(db: Session, transaction_id: str):
+    transaction_history_to_delete = db.query(models.TransactionHistory).filter(models.TransactionHistory.transaction_id == transaction_id).first()
+    db.delete(transaction_history_to_delete)
+    db.commit()
+
+# transaction_items
+def create_transaction_item(db: Session, transaction_item = schemas.TransactionItemCreate):
+    db_transaction_item = models.TransactionItem(
+        item_id = transaction_item.item_id,
+        transaction_id = transaction_item.transaction_id,
+        transaction_quantity = transaction_item.transaction_quantity
+    )
+    db.add(db_transaction_item)
+    db.commit()
+    db.refresh(db_transaction_item)
+
+def get_transaction_items_by_transaction_id(db: Session, transaction_id: str):
+    return db.query(models.TransactionItem).filter(models.TransactionItem.transaction_id == transaction_id).all()
+
+def delete_transaction_items_by_item_id(db: Session, item_id: str):
```

Comment:
Combine this with the function starting on line 206, similar to the replenishment. Take an item_id and a transaction_id